clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeCheckerTest.java

```diff
@@ -139,6 +146,158 @@ public void testUnknownStorageNode() {
         assertThat(result.getReason(), is("Unknown node storage.10"));
     }
 
+    @Test
+    public void testWhenOtherStorageNodeIsSuspended() {
+        // Nodes 0-3, storage node 0 being in maintenance with "Orchestrator" description.
+        ContentCluster cluster = createCluster(createNodes(4));
+        cluster.clusterInfo().getStorageNodeInfo(0).setWantedState(new NodeState(NodeType.STORAGE, State.MAINTENANCE).setDescription("Orchestrator"));
+        NodeStateChangeChecker nodeStateChangeChecker = new NodeStateChangeChecker(
+                requiredRedundancy, noopVisiting, cluster.clusterInfo(), false);
+        ClusterState clusterStateWith0InMaintenance = clusterState(String.format(
+                "version:%d distributor:4 storage:4 .0.s:m",
+                currentClusterStateVersion));
+
+        NodeStateChangeChecker.Result result = nodeStateChangeChecker.evaluateTransition(
+                new Node(NodeType.STORAGE, 1), clusterStateWith0InMaintenance,
+                SetUnitStateRequest.Condition.SAFE, UP_NODE_STATE, MAINTENANCE_NODE_STATE);
+        assertFalse(result.settingWantedStateIsAllowed());
+        assertFalse(result.wantedStateAlreadySet());
+        assertThat(result.getReason(), is("At most one node can have a wanted state when #groups = 1: " +
+                "Other storage node 0 has wanted state Maintenance"));
+    }
+
+    @Test
+    public void testWhenOtherDistributorIsDown() {
+        // Nodes 0-3, storage node 0 being in maintenance with "Orchestrator" description.
+        ContentCluster cluster = createCluster(createNodes(4));
+        cluster.clusterInfo().getDistributorNodeInfo(0)
+                .setWantedState(new NodeState(NodeType.DISTRIBUTOR, State.DOWN).setDescription("Orchestrator"));
```

Comment:
Should we test without the explicit `Orchestrator` reason for at least one test? I seem to recall we have some special logic for that particular string.

Reply:
"Orchestrator" isn't hard-coded in the cluster controller, but the rule "allow suspending if another node in the group is suspended" finds Maintenance nodes with the same description, hence would find "Orchestrator" Maintenance storage nodes.  This match on the description is tested in testDifferentDescriptionImpliesDenied.

The description is set to Orchestrator here mostly for documentation purposes - I could remove it or use a random description.